src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1

```diff
@@ -286,27 +372,80 @@ class AzSKADOServiceMapping: CommandBase
             if (([Helpers]::CheckMember($buildDefnsObj, "count") -and $buildDefnsObj[0].count -gt 0) -or (($buildDefnsObj | Measure-Object).Count -gt 0 -and [Helpers]::CheckMember($buildDefnsObj[0], "name"))) {
 
                 $this.PublishCustomMessage(([Constants]::DoubleDashLine))
-                $this.PublishCustomMessage("Generating service mappings of variable group using build for project [$($this.ProjectName)]...")
+                $this.PublishCustomMessage("Generating service mappings of variable group/secure file using build for project [$($this.ProjectName)]...")
                 $this.PublishCustomMessage("Total mappings to be evaluated:  $(($buildDefnsObj | Measure-Object).Count)")
                 $counter = 0
 
                 foreach ($bldDef in $buildDefnsObj) {
                     $counter++
-                    Write-Progress -Activity 'Variable group mappings via build...' -CurrentOperation $bldDef.Name -PercentComplete (($counter / $buildDefnsObj.count) * 100)
+                    Write-Progress -Activity 'Variable group/secure file mappings via build...' -CurrentOperation $bldDef.Name -PercentComplete (($counter / $buildDefnsObj.count) * 100)
 
                     $buildObj = [WebRequestHelper]::InvokeGetWebRequest($bldDef.url.split('?')[0]);
 
-                    if([Helpers]::CheckMember($buildObj[0],"variableGroups"))
-                    {
-                        $varGrps = $buildObj[0].variableGroups
-                        $varGrps | ForEach-Object{
+                    #getting secure files added in all the tasks.
+                    try {
```

Comment:
Is try catch needed here?

Reply:
yes, we are drilling down, the objects property , checkmember is there, but let try catch be there for safety if any exceptional pipeline breaks , then it does not break loop